aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/filters/UrlPathValidator.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.amazonaws.serverless.proxy.internal.servlet.filters;
+
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+/**
+ * Simple path validator filter. This is a default implementation to prevent malformed paths from hitting the framework
+ * app. This applies to all paths by default
+ */
+@WebFilter(filterName = "UrlPathValidator", urlPatterns = {"/*"})
+public class UrlPathValidator implements Filter {
+
+    //-------------------------------------------------------------
+    // Constants
+    //-------------------------------------------------------------
+
+    public static final int DEFAULT_ERROR_CODE = 404;
+    public static final Pattern PATH_PATTERN = Pattern.compile("^(/[-\\w:@&?=+,.!/~*'%$_;]*)?$");
+
+
+    //-------------------------------------------------------------
+    // Variables - Private
+    //-------------------------------------------------------------
+
+    private int invalidStatusCode;
+
+
+    //-------------------------------------------------------------
+    // Implementation - Filter
+    //-------------------------------------------------------------
+
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        if (filterConfig.getInitParameter("invalid_status_code") != null) {
+            String statusCode = filterConfig.getInitParameter("invalid_status_code");
+            try {
+                invalidStatusCode = Integer.parseInt(statusCode);
+            } catch (NumberFormatException e) {
+                invalidStatusCode = DEFAULT_ERROR_CODE;
```

Comment:
Need to find a better log strategy. Tired of just calling the lambdaContext.getLogger from places I can't reach it. I'll add this as an issue for the next release.